### PR TITLE
Refactor signal and command line handler from resident runner

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Initial work for https://github.com/flutter/flutter/issues/34552

Resident/Hot/Cold runners have both too many responsibilities and are under-tested compared to other parts of the code base.

This is an attempt at a light refactoring, only removing the terminal and signal handling capability to a new class `TerminalHandler` which must be provided with a ResidentRunner instance. Also added a simple smoke test of attach from the hot runner, and a small fix to consistency of the member `usesTerminalUi`.

### CODE REVIEW

File: packages/flutter_tools/test/resident_runner_test.dart
Comment (dnfield) [inline packages/flutter_tools/test/resident_runner_test.dart]: Why are we changing the date?

**Review by dnfield — APPROVED** [review]
LGTM modulo date change

**Comment by dnfield** [comment]
My other change landed, will be happy to re-review once that's merged into this.

File: packages/flutter_tools/lib/src/resident_runner.dart
Comment (dnfield) [inline packages/flutter_tools/lib/src/resident_runner.dart]: Can we turn this into the switch case statement?  I think it will make reading this easier.

Comment (jonahwilliams) [reply packages/flutter_tools/lib/src/resident_runner.dart]: Done

**Review by dnfield — APPROVED** [review]

**Comment by jonahwilliams** [comment]
Ignoring codecov patch until I figure out why it still doesn't like the diff...

**Pull request merged**
